components/public-api-server/pkg/server/integration_test.go

```diff
@@ -13,151 +13,101 @@ import (
+	requireErrorStatusCode(t, connect.CodeUnimplemented, err)
 
-	stopWorkspaceStream, err := workspaceClient.StopWorkspace(ctx, &v1.StopWorkspaceRequest{})
+	stopWorkspaceStream, err := workspaceClient.StopWorkspace(ctx, connect.NewRequest(&v1.StopWorkspaceRequest{}))
 	require.NoError(t, err)
-	_, err = stopWorkspaceStream.Recv()
-	requireErrorStatusCode(t, codes.Unimplemented, err)
+	_ = stopWorkspaceStream.Receive()
+	err = stopWorkspaceStream.Err()
+	requireErrorStatusCode(t, connect.CodeUnimplemented, err)
 
-	listenWorkspaceStream, err := workspaceClient.ListenToWorkspaceInstance(ctx, &v1.ListenToWorkspaceInstanceRequest{})
+	listenWorkspaceStream, err := workspaceClient.ListenToWorkspaceInstance(ctx, connect.NewRequest(&v1.ListenToWorkspaceInstanceRequest{}))
 	require.NoError(t, err)
-	_, err = listenWorkspaceStream.Recv()
-	requireErrorStatusCode(t, codes.Unimplemented, err)
+	_ = listenWorkspaceStream.Receive()
+	err = listenWorkspaceStream.Err()
+	requireErrorStatusCode(t, connect.CodeUnimplemented, err)
 
-	listenImageBuildStream, err := workspaceClient.ListenToImageBuildLogs(ctx, &v1.ListenToImageBuildLogsRequest{})
+	listenImageBuildStream, err := workspaceClient.ListenToImageBuildLogs(ctx, connect.NewRequest(&v1.ListenToImageBuildLogsRequest{}))
 	require.NoError(t, err)
-	_, err = listenImageBuildStream.Recv()
-	requireErrorStatusCode(t, codes.Unimplemented, err)
+	_ = listenImageBuildStream.Receive()
+	err = listenImageBuildStream.Err()
+	requireErrorStatusCode(t, connect.CodeUnimplemented, err)
 }
 
 func TestPublicAPIServer_v1_PrebuildService(t *testing.T) {
 	ctx := context.Background()
-	srv := baseserver.NewForTests(t, baseserver.WithGRPC(baseserver.MustUseRandomLocalAddress(t)))
+	srv := baseserver.NewForTests(t, baseserver.WithHTTP(baseserver.MustUseRandomLocalAddress(t)))
 
 	gitpodAPI, err := url.Parse("wss://main.preview.gitpod-dev.com/api/v1")
 	require.NoError(t, err)
 
-	require.NoError(t, register(srv, gitpodAPI))
-
-	baseserver.StartServerForTests(t, srv)
-
-	conn, err := grpc.Dial(srv.GRPCAddress(), grpc.WithTransportCredentials(insecure.NewCredentials()))
-	require.NoError(t, err)
-
-	prebuildClient := v1.NewPrebuildsServiceClient(conn)
-
-	_, err = prebuildClient.GetPrebuild(ctx, &v1.GetPrebuildRequest{})
-	requireErrorStatusCode(t, codes.Unimplemented, err)
-
-	_, err = prebuildClient.GetRunningPrebuild(ctx, &v1.GetRunningPrebuildRequest{})
-	requireErrorStatusCode(t, codes.Unimplemented, err)
-
-	listenToStatusStream, err := prebuildClient.ListenToPrebuildStatus(ctx, &v1.ListenToPrebuildStatusRequest{})
-	require.NoError(t, err)
-	_, err = listenToStatusStream.Recv()
-	requireErrorStatusCode(t, codes.Unimplemented, err)
-
-	listenToLogsStream, err := prebuildClient.ListenToPrebuildLogs(ctx, &v1.ListenToPrebuildLogsRequest{})
-	require.NoError(t, err)
-	_, err = listenToLogsStream.Recv()
-	requireErrorStatusCode(t, codes.Unimplemented, err)
-}
-
-func TestPublicAPIServer_WorkspaceServiceHandler(t *testing.T) {
```

Comment:
Diff shows wierd. This test is removed as it's already tested in `TestPublicAPIServer_v1_WorkspaceService`